src/shared/previewUtils.ts

```diff
@@ -30,14 +30,51 @@ import {
 } from '@salesforce/lwc-dev-mobile-core';
 import { Progress, Spinner } from '@salesforce/sf-plugins-core';
 import fetch from 'node-fetch';
+import { SecureConnectionFiles } from '../configMeta.js';
+import { ConfigUtils, LOCAL_DEV_SERVER_DEFAULT_PORT } from './configUtils.js';
 
 Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
 const messages = Messages.loadMessages('@salesforce/plugin-lightning-dev', 'lightning.preview.app');
 const DevPreviewAuraMode = 'DEVPREVIEW';
 
 export class PreviewUtils {
-  public static generateWebSocketUrlForLocalDevServer(platform: string, port: string): string {
-    return LwcDevMobileCorePreviewUtils.generateWebSocketUrlForLocalDevServer(platform, port);
+  public static generateWebSocketUrlForLocalDevServer(platform: string, port: string | number): string {
+    return LwcDevMobileCorePreviewUtils.generateWebSocketUrlForLocalDevServer(platform, port.toString());
+  }
+
+  /**
+   * Return a port number to be used by the local dev server.
+   *
+   * It starts with the default port (8081)) and checks to see if it is in use or not. If
+   * it is in use then we increment the port number by 2 and check if it is in use or not.
+   * This process is repeated until a port that is not in use is found.
+   *
+   * @returns a port number to be used by the local dev server.
+   */
```

Comment:
With this approach, we are able to preview multiple SFDX projects at the same time (if customers ever need to) and have each of them use a different port